src/main/java/org/obsidiantoaster/generator/rest/ObsidianResource.java

```diff
@@ -293,20 +344,105 @@ public Response downloadZip(JsonObject content,
       }
    }
 
-   protected void validateCommand(String commandName)
+   @POST
+   @javax.ws.rs.Path("/commands/{commandName}/execute")
+   @Consumes(MediaType.APPLICATION_JSON)
+   public Response executeCommandJson(JsonObject content,
+            @PathParam("commandName") @DefaultValue(DEFAULT_COMMAND_NAME) String commandName,
+            @Context HttpHeaders headers)
+            throws Exception
    {
-      if (commandMap.get(commandName) == null)
+      exposedCommands.validateCommand(commandName);
+      try (CommandController controller = getCommand(commandName, headers))
       {
-         String message = "No such command `" + commandName + "`. Supported commmands are '"
-                  + String.join("', '", commandMap.keySet()) + "'";
-         throw new WebApplicationException(message, Status.NOT_FOUND);
+         helper.populateControllerAllInputs(content, controller);
+         if (controller.isValid())
+         {
+            Result result = controller.execute();
+            if (result instanceof Failed)
+            {
+               JsonObjectBuilder builder = Json.createObjectBuilder();
+               helper.describeResult(builder,result);
+               return Response.status(Status.INTERNAL_SERVER_ERROR).entity(builder).build();
+            }
+            else
+            {
+               Object entity = getEntity(result);
+               if (entity != null)
+               {
+                  return Response
+                           .ok(entity)
+                           .type(MediaType.APPLICATION_JSON)
+                           .build();
+               }
+               else
+               {
+                  return Response
+                           .ok(getMessage(result))
+                           .type(MediaType.TEXT_PLAIN)
+                           .build();
+               }
+            }
+         }
+         else
+         {
+            JsonObjectBuilder builder = createObjectBuilder();
+            helper.describeValidation(builder, controller);
+            return Response.status(Status.PRECONDITION_FAILED).entity(builder.build()).build();
+         }
+      }
+   }
+
+   /**
+    * Returns the entity from the result handling {@link CompositeResult} values as a List of entities
+    */
+   protected static Object getEntity(Result result)
```

Comment:
This method is not really clear on it's return type. It can return a List or any object?